### ISSUE TITLE
v4 broke AMD default object export

### ISSUE THREAD

**Issue by kouts** [systemjs/systemjs#1967]
After upgrading from v3.1.6 to 4.0.2 AMD modules that export an object no longer work.
Example using Babel @babel/plugin-transform-modules-amd
```
define(["exports", "main"], function (_exports, _main) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports["default"] = void 0;
  var _default = {
    test: 'hello world'
  };
  _exports["default"] = _default;
});
```
In order to be used it must be consumed as ```module.default.default``` instead of ```module.default```

**Comment by guybedford** [comment]
This could well be a bug.

How are you importing the module? When used from another AMD module it should be `exports.default` but when imported via `System.import` it should be `exports.default.default`.

**Comment by kouts** [comment]
I'm importing it via ```System.import``` e.g
```
System.import('amd_test_object.js').then(m => {
    // Worked with v 3.1.6
    console.log(m.default);
    // in order to work with v 4.0.2
    console.log(m.default.default);
});
```

**Comment by joeldenning** [comment]
From the 4.0.0 changelog:

> Support AMD importing ESM with an internal interop signifier (https://github.com/systemjs/systemjs/pull/1941)

^^ This is what caused the change. It is intentional. See https://github.com/systemjs/systemjs/issues/1933 for the motivation and discussion behind this change.

Having to do `.default.default` is not ideal. Some history behind what is happening: `@babel/plugin-transform-amd`'s decision to do `exports['default'] = _default` instead of `exports = _default` is related to a proposal/convention for `__esModule` that was adopted a few years ago by Webpack, Babel, SystemJS, etc. @guybedford was the one who proposed it and it became ubiquitous. However, more recently the `__esModule` property has lost favor and is not going to be supported in NodeJS for ESM interop. It is a conscious decision for SystemJS@>=3 not to support `__esModule` stuff so that it behaves similarly to the future of the javascript ecosystem.

@guybedford feel free to jump in and clarify anything I missed -- the above is my understanding of how we got to a place where `.default.default` is needed 😄

**Comment by joeldenning** [comment]
Perhaps the creation of an `__esModule.js` [extra](https://github.com/systemjs/systemjs#extras) would be helpful for @kouts and for those in a similar situation? One where people could opt into the older `__esModule` behavior if they really want it?

**Comment by guybedford** [comment]
Right, yes this is intentional because the module really is a `.default.default` (eg if you compiled to CommonJS and loaded in `--experimental-modules` that is what you would get!).

An `__esModule` extra may well make sense, if we can do this as an extra without needing any further core hooks (I'm hesitant to bring anything around this into core though).

**Comment by joeldenning** [comment]
Would it be an __esModule extra that would help here? Or a `__useDefault` extra?

```js
// An __esModule extra?
const originalImport = System.import;
System.import = function() {
  return originalImport.apply(this, arguments).then(function (module) {
    return module.__esModule ? module.default : module;
  });
};
```

```js
// A __useDefault extra?
const originalImport = System.import;
System.import = function() {
  return originalImport.apply(this, arguments).then(function (module) {
    return module.__useDefault ? module.default : module;
  });
};
```

I feel like the __esModule might not actually make sense, since __esModule is something that you have to do *after* you get the module and know which imports you care about. For example, webpack's interopRequireDefault function does the __esModule check **after** getting an entire external module from SystemJS.

Maybe the __useDefault extra makes more sense?

**Comment by guybedford** [comment]
@joeldenning certainly a `System.import` `__useDefault` extra makes sense.

**Comment by kouts** [comment]
Bear in mind this happens with AMD modules.
If I compile the same code as a SystemJS module (using @babel/plugin-transform-modules-systemjs)
I can consume it as ```module.default``` as before.

**Comment by joeldenning** [comment]
Released in 4.1.0
